web3/pm.py

```diff
@@ -1,27 +1,220 @@
         "`ethpm` dependency manually: pip install --upgrade ethpm"
     ) from exc
 
 
-# Package Management is currently still in alpha
+# Package Management is currently still in alpha.
 # It is not automatically available on a web3 object.
 # To use the `PM` module, attach it to your web3 object
 # i.e. PM.attach(web3, 'pm')
+
+
 class PM(Module):
     def get_package_from_manifest(self, manifest):
+        """
+        * :manifest: must be a dict representing a valid manifest
+        * Returns a ``Package`` instance representing the Manifest
+        """
         pkg = Package(manifest, self.web3)
         return pkg
 
-    def get_package_from_uri(self, uri):
-        pkg = Package.from_uri(uri, self.web3)
+    def get_package_from_uri(self, manifest_uri):
+        """
+        * :uri: *must* be a valid content-addressed URI, as defined in the
+        `Py-EthPM Documentation <https://py-ethpm.readthedocs.io/en/latest/uri_backends.html>`_.
+        * Returns a ``Package`` isntance representing the Manifest stored at the URI.
+        """
+        pkg = Package.from_uri(manifest_uri, self.web3)
         return pkg
+
+    def set_registry(self, address, ens_address=None):
+        """
+        Sets the current registry used in ``web3.pm`` functions that read/write to an
+        onchain registry.
+
+        :address: Address of the on-chain registry - Accepts ENS name
+        :ens_address: If using an ens, but not on the mainnet, this is the address of the central
+        ENS registry on whatever chain being used.
+        """
+        if is_canonical_address(address):
+            self.registry = Registry(address, self.web3)
+        elif isinstance(address, str):
+            ns = ENS.fromWeb3(self.web3, ens_address)
+            addr_lookup = ns.address(address)
+            if addr_lookup:
+                self.registry = Registry(addr_lookup, self.web3)
+            else:
+                raise PMError("No address found after ENS lookup for name: {0}.".format(address))
+        else:
+            raise PMError(
+                "Expected an address or ENS name for the address, "
+                "instead received {0}.".format(type(address))
+            )
+
+    def deploy_and_set_registry(self):
+        """
+        Deploys a new instance of a Registry.vy and sets that Registry to ``web3.pm``.
+        (py-ethpm/ethpm/assets/vyper_registry/registry.v.py)
+        """
+        self.registry = Registry.deploy_new_instance(self.web3)
+
+    def tie_registry_to_ens(self, ens_name, ens_address=None):
+        """
+        Sets ``web3.PM`` current registry to the given ``ens_name``. Provide an ``ens_address``
+        if you're using an ENS not on the mainnet. ``web3.eth.defaultAccount`` must be an
+        owner of the domain name for this method to work.
+        """
+        self._validate_set_registry()
+        ns = ENS.fromWeb3(self.web3, ens_address)
+        ns.setup_address(ens_name, self.registry.address)
```

Comment:
This seems like a simple enough operation that the whole method doesn't need to exist. It's a one-liner: `w3.ens.setup_address(ens_name, w3.pm.registry.address)`